store/src/java/com/zimbra/cs/account/AuthToken.java

```diff
@@ -210,6 +211,16 @@ public abstract void encode(HttpServletResponse resp, boolean isAdminReq,
             boolean secureCookie, boolean rememberMe)
     throws ServiceException;
 
+    public void encode(HttpServletRequest reqst, HttpServletResponse resp, boolean isAdminReq, boolean secureCookie, boolean rememberMe)
+    throws ServiceException {
+        encode(resp, isAdminReq, secureCookie, rememberMe);
+    }
+
+    public void encode(HttpServletRequest reqst, HttpServletResponse resp, boolean deregister)
+    throws ServiceException {
+        //do nothing
```

Comment:
might be cleaner to keep this abstract and overwrite in ZimbraAuthToken to do whatever it needs to (or nothing)

Reply:
This method is needed for ZimbraJWToken class which overrides this method. At present, it has no significance for other implementations of AuthToken class. if any implementation want to use this method, they can override it. 
Can't keep this method abstract in AuthToken as that would force all existing implementations of AuthToken to implement this method and it would break code of other people who have overridden AuthToken class.

Reply:
Please add a comment explaining the utility of the method - perhaps what you've written here would be good content :)

Reply:
comment added.